cmd/root.go

```diff
@@ -454,21 +417,60 @@ func expandEnvCmdObjs(cmd *cobra.Command, envSpec *envSpec, cwd metadata.AbsPath
 //      foo: import "components/foo.jsonnet"
 //      "foo-bar": import "components/foo-bar.jsonnet"
 //   }
-func constructBaseObj(paths []string) string {
+func constructBaseObj(componentPaths, componentNames []string) (string, error) {
+	// IMPLEMENTATION NOTE: If one or more `componentNames` exist, it is
+	// sufficient to simply omit every name that does not appear in the list. This
+	// is because we know every field of the base object will contain _only_ an
+	// `import` node (see example object in the function-heading comment). This
+	// would not be true in cases where one field can reference another field; in
+	// this case, one would need to generate the entire object, and filter that.
+	//
+	// Hence, a word of caution: if the base object ever becomes more complex, you
+	// will need to change the way this function performs filtering, as it will
+	// lead to very confusing bugs.
+
+	filter := map[string]string{}
+	for _, name := range componentNames {
+		filter[name] = ""
+	}
+
+	// Add every component we know about to the base object.
 	var obj bytes.Buffer
 	obj.WriteString("{\n")
-	for _, p := range paths {
+	for _, p := range componentPaths {
 		ext := path.Ext(p)
-		if path.Ext(p) != ".jsonnet" {
+
+		// Generate import statement.
+		var importExpr string
+		switch ext {
+		case ".jsonnet":
+			importExpr = fmt.Sprintf(`import "%s"`, p)
+		case ".yaml", ".yml":
+			importExpr = fmt.Sprintf(`kubecfg.parseYaml(%s)`, p)
+		case ".json":
+			importExpr = fmt.Sprintf(`kubecfg.parseJson(%s)`, p)
+		default:
 			continue
 		}
 
 		name := strings.TrimSuffix(path.Base(p), ext)
 		name = params.SanitizeComponent(name)
-		fmt.Fprintf(&obj, "  %s: import \"%s\",\n", name, p)
+
+		// Filter! If the filter has more than 1 element and the component name is
+		// not in the filter, skip.
+		if _, exists := filter[name]; len(filter) > 0 && !exists {
```

Comment:
The `len(filter) > 0` check here is a little confusing to me. Can it be pulled out, before the loop?

Can we also put this specific logic at the top of the loop -- so that we can ignore all the generating import statement logic if the check passes?